packages/application-generic/src/services/cache/key-builders/entities.ts

```diff
@@ -1,173 +1,176 @@
+export const buildAuthServiceKey = ({ apiKey }: { apiKey: string }): string => {
   const apiKeyHash = createHash(apiKey);
 
-  return buildKeyById({
+  return buildUnscopedKey({
     type: CacheKeyTypeEnum.ENTITY,
     keyEntity: CacheKeyPrefixEnum.AUTH_SERVICE,
     identifier: apiKeyHash,
     identifierPrefix: IdentifierPrefixEnum.API_KEY,
   });
 };
 
-const buildMaximumApiRateLimitKey = ({
+export const buildMaximumApiRateLimitKey = ({
   apiRateLimitCategory,
   _environmentId,
 }: {
   apiRateLimitCategory: string;
   _environmentId: string;
 }): string =>
-  buildCommonKey({
+  buildEnvironmentScopedKeyById({
     type: CacheKeyTypeEnum.ENTITY,
     keyEntity: CacheKeyPrefixEnum.MAXIMUM_API_RATE_LIMIT,
     environmentId: _environmentId,
     identifierPrefix: IdentifierPrefixEnum.API_RATE_LIMIT_CATEGORY,
     identifier: apiRateLimitCategory,
   });
 
-const buildEvaluateApiRateLimitKey = ({
+export const buildEvaluateApiRateLimitKey = ({
   apiRateLimitCategory,
   _environmentId,
 }: {
   apiRateLimitCategory: string;
   _environmentId: string;
 }): string =>
-  buildCommonKey({
+  buildEnvironmentScopedKeyById({
     type: CacheKeyTypeEnum.ENTITY,
     keyEntity: CacheKeyPrefixEnum.EVALUATE_API_RATE_LIMIT,
     environmentId: _environmentId,
     identifierPrefix: IdentifierPrefixEnum.API_RATE_LIMIT_CATEGORY,
     identifier: apiRateLimitCategory,
   });
 
-const buildServiceConfigKey = (
-  identifier: ServiceConfigIdentifierEnum
-): string =>
-  buildKeyById({
+export const buildEventUsageKey = ({
+  _organizationId,
+  resourceType,
+  periodStart,
+  periodEnd,
+}: {
+  _organizationId: string;
+  resourceType: string;
+  periodStart: number;
+  periodEnd: number;
+}): string =>
+  buildOrganizationScopedKeyById({
+    type: CacheKeyTypeEnum.ENTITY,
+    keyEntity: CacheKeyPrefixEnum.USAGE,
+    identifierPrefix: IdentifierPrefixEnum.RESOURCE_TYPE,
+    identifier: `${resourceType}_${periodStart}_${periodEnd}`,
+    organizationId: _organizationId,
+  });
+
+export const buildSubscriptionKey = ({
+  organizationId,
+}: {
+  organizationId: string;
+}): string =>
+  buildOrganizationScopedKey({
     type: CacheKeyTypeEnum.ENTITY,
-    keyEntity: CacheKeyPrefixEnum.SERVICE_CONFIG,
-    identifierPrefix: IdentifierPrefixEnum.SERVICE_CONFIG,
-    identifier,
+    keyEntity: CacheKeyPrefixEnum.SUBSCRIPTION,
+    organizationId,
   });
 
-const buildServiceConfigApiRateLimitMaximumKey = (): string =>
+export const buildServiceConfigApiRateLimitMaximumKey = (): string =>
   buildServiceConfigKey(
     ServiceConfigIdentifierEnum.API_RATE_LIMIT_SERVICE_MAXIMUM
   );
-
-export {
-  buildUserKey,
-  buildSubscriberKey,
-  buildNotificationTemplateKey,
-  buildNotificationTemplateIdentifierKey,
-  buildEnvironmentByApiKey,
-  buildKeyById,
-  buildGroupedBlueprintsKey,
-  buildAuthServiceKey,
-  buildMaximumApiRateLimitKey,
-  buildEvaluateApiRateLimitKey,
-  buildServiceConfigApiRateLimitMaximumKey,
-  buildVariablesKey,
-};
```

Comment:
It's a bit of a code smell to update here, so we instead export each of the required builders in-line.